src/states/elder/mod.rs

```diff
@@ -1582,53 +1586,35 @@ impl Approved for Elder {
         &mut self,
         elders_info: EldersInfo,
         old_pfx: Prefix<XorName>,
-        neighbour_change: EldersChange,
         outbox: &mut dyn EventBox,
     ) -> Result<Transition, RoutingError> {
         info!("{} - handle SectionInfo: {:?}.", self, elders_info);
 
-        let self_sec_update = elders_info.prefix().matches(self.name());
-
         // Poll the relocate queue before the parsec reset, so it is not blocked waiting for the
         // genesis event. Cast the actual votes only after the parsec reset however, so they already
         // go to the new instance.
-        let relocate_details = if self_sec_update {
-            self.chain.poll_relocation()
-        } else {
-            None
-        };
+        let relocate_details = self.chain.poll_relocation();
 
         if elders_info.prefix().is_extension_of(&old_pfx) {
-            self.finalise_prefix_change()?;
-            self.send_event(Event::SectionSplit(*elders_info.prefix()), outbox);
-            // After a section split, the normal `send_neighbour_infos` action for the neighbouring
-            // section will be triggered here (and only here).  Meanwhile own section's sending
-            // action will be triggered at the other place later on (`self_sec_update` is true).
-            if !elders_info.prefix().matches(self.name()) {
-                self.send_neighbour_infos();
-            }
+            self.finalise_split(outbox)?;
         } else if old_pfx.is_extension_of(elders_info.prefix()) {
             panic!(
                 "{} - Merge not supported: {:?} -> {:?}",
                 self,
                 old_pfx,
                 elders_info.prefix()
             );
-        } else if self_sec_update {
+        } else {
             self.reset_parsec()?;
         }
 
-        self.update_neighbour_connections(neighbour_change, outbox);
```

Comment:
I am not 100% sure we can remove this `update_neighbour_connections`, because it does establish connections with other elders as well, but it is unlikely to be an issue.